File: Pythonleetcode/Leetcode63.py
class Solution(object):
    def uniquePathsWithObstacles(self, obstacleGrid):
        """
        :type obstacleGrid: List[List[int]]
        :rtype: int
        """
        dp=[[0 for i in range(len(obstacleGrid[0]))] for j in range(len(obstacleGrid))]
        for i in range(len(obstacleGrid)):
            for j in range(len(obstacleGrid[0])):
                if obstacleGrid[i][j]==1:
                    dp[i][j]=0
                elif j==0 and i==0:
                    dp[i][j]=1
                else:
                    if i>0:
                        dp[i][j]+=dp[i-1][j]
                    if j>0:
                        dp[i][j]+=dp[i][j-1]
        return dp[-1][-1]

print(Solution().uniquePathsWithObstacles([[0,0,0],[0,1,0],[0,0,0]]))
print(Solution().uniquePathsWithObstacles([[1,0,0]]))

File: CSharpPractice/LeetcodePractice/Leetcode63.cs
public class Solution63 {
    public int UniquePathsWithObstacles(int[][] obstacleGrid) 
    {
        int m = obstacleGrid.Length;
        int n = obstacleGrid[0].Length;
        int[,] dp = new int[m,n];
        if (obstacleGrid[0][0]==0)
        {
            dp[0,0] = 1;
        }
        for (var i = 1; i < m; i++)
        {
            if (obstacleGrid[i][0]==0)
            {
                dp[i,0] = dp[i-1,0];
            }
        }
        for (var j = 1; j < n; j++)
        {
            if (obstacleGrid[0][j]==0)
            {
                dp[0,j] = dp[0,j-1];
            }
        }
        for (var i = 1; i < m; i++)
        {
            for (var j = 1; j < n; j++)
            {
                if (obstacleGrid[i][j]==0)
                {
                    dp[i,j] = dp[i-1,j]+dp[i,j-1];
                }
            }
        }
        return dp[m-1,n-1];
    }
}